src/Decompiler/Analysis/UsedRegisterFinder.cs

```diff
@@ -213,8 +213,7 @@ SsaIdentifier SidOf(PhiArgument a)
                 }
             }
             typeDescender.MergeDataTypes();
-            return typeDescender.TypeVariables[sidParam.Identifier]
-                .DataType;
+            return typeDescender.GetType(sidParam.Identifier);
```

Comment:
An idea here, not necessarily needed to be implemented immediately. Instead of a method `GetType` on `TypeDescender` and subclasses, consider retrieving the type variables of expressions by calling `TypeStore.GetTypeVariable`

Reply:
I can't get it. Do you mean refactor  `LocalTypeDescender` to remove `TypeVariables` map? `GetType` method is still needed in this case